crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1813,6 +1807,46 @@ mod tests {
         assert_eq!(DROP_COUNT.load(std::sync::atomic::Ordering::SeqCst), 1);
     }
 
+    #[test]
+    fn insert_resource_by_id_drop_old() {
```

Comment:
You should use `DropTestHelper` and a regular rust type for the resource instead of doing this all dynamically. You can get the `ComponentId` representing the rust type via `world.components.get_resource_id(TypeId::of::<Whatever>()).unwrap()` (after you've inserted the resource the first time otherwise it wont have been registered lol)

Reply:
> Alternatively I can create the world inside the catch_unwind and not remove the !contains_resource assert.

I prefer this approach as it's a bit less spooky.

Reply:
You can set `make_component(false, 1)` and get rid of the catch unwind/assertunwindsafe. I don't think panic in drop was relevent to this issue was it?

Reply:
Oh thats unfortunate oh well ...